### ISSUE TITLE
UI elements for addon layer checkboxes and addon buttons duplicate when switching map type

### ISSUE THREAD

**Issue by Adambean** [awakening-wurm/MapViewer#3]
Hello,

I've began to use this map viewer in replacement of another. (Completely fresh installation.) So far it's looking splendid.

I noticed when switching between the 3 map types that the addon layer checkboxes (highways, kingdoms, and locate soul assistant) duplicate, as do their informational buttons in the bottom right corner. Only the 3 new duplicates of the layer checkboxes work. The ones that already exist no longer function at all, so they're probably just not getting removed prior to refreshing the UI.

This can be tried out here: https://games.reece-eu.net/wurm/maps/Creative-Beanland
Version is 1.2.1.

**Comment by perlowgren** [comment]
Please clear browser cache and reload and see if it solvers it. Your map is working in my browser.

**Comment by Adambean** [comment]
Thanks for the quick response.
Cleared Firefox's cache, this still happens. First visit on Chrome also has this. I've also tried in Edge (which I haven't used for months, and definitely never for this map), it happens on that too.

![image](https://user-images.githubusercontent.com/1108717/129107542-697201aa-0d8f-437a-8e8a-04c7fde75461.png)
![image](https://user-images.githubusercontent.com/1108717/129107576-33f37361-c3c1-4323-b13e-2b7130ef00f3.png)

I'll ask a few other players if this happens to them too.

**Comment by perlowgren** [comment]
Oh, I re-read your post, now I see what you mean. Yeah, I can fix this bug, it's when you switch map types this only happens, it's because the "mapLoaded" event is triggered each time the map loads. Instead I'll trigger the event only once.

**Comment by perlowgren** [comment]
Thanks for reporting!

**Comment by Adambean** [comment]
Thanks for looking into this, I'll check out v1.2.2 this evening. :)
